src/handlers/v1/markets/[chainId]/[marketId]/zap-repay/index.ts

```diff
@@ -0,0 +1,161 @@
+
+export const v1GetZapRepayQuotationRoute: Route<GetZapRepayQuotationRouteParams> = {
+  method: 'POST',
+  path: '/v1/markets/{chainId}/{marketId}/zap-repay',
+  handler: async (event) => {
+    const chainId = Number(event.pathParameters.chainId);
+    const marketId = event.pathParameters.marketId.toUpperCase();
+    if (!validateMarket(chainId, marketId)) {
+      throw newHttpError(400, { code: '400.1', message: 'market does not exist' });
+    }
+
+    if (!event.body) {
+      throw newHttpError(400, { code: '400.2', message: 'body is invalid' });
+    }
+
+    let account = event.body.account;
+    if (!account) {
+      throw newHttpError(400, { code: '400.3', message: `account can't be blank` });
+    }
+    try {
+      account = utils.getAddress(account);
+    } catch {
+      throw newHttpError(400, { code: '400.4', message: 'account is invalid' });
+    }
+
+    const service = new Service(chainId);
+
+    let marketInfo: MarketInfo;
+    try {
+      marketInfo = await service.getMarketInfo(marketId, account);
+    } catch (err) {
+      throw newInternalServerError(err);
+    }
+    const { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR } = marketInfo;
+    const currentPosition = { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR };
+
+    let targetTokenAmount = '0';
+    const logics: GetZapRepayQuotationResponseBody['logics'] = [];
+    let fees: GetZapRepayQuotationResponseBody['fees'] = [];
+    let approvals: GetZapRepayQuotationResponseBody['approvals'] = [];
+    let permitData: GetZapRepayQuotationResponseBody['permitData'];
+    let targetPosition = currentPosition;
+    if (event.body.sourceToken && event.body.amount && Number(event.body.amount) > 0) {
+      const { amount, slippage } = event.body;
+      const { baseToken, supplyAPR, supplyUSD, borrowAPR, borrowCapacityUSD, liquidationLimit, baseTokenPrice } =
+        marketInfo;
+      const sourceToken = common.Token.from(event.body.sourceToken);
+
+      // 1. check borrow USD
+      if (new BigNumberJS(borrowUSD).isZero()) {
+        throw newHttpError(400, { code: '400.5', message: 'borrow USD is zero' });
+      }
+
+      // 2. new and append swap token logic
+      let repayToken: common.Token;
+      if (sourceToken.unwrapped.is(baseToken)) {
+        repayToken = sourceToken;
+        targetTokenAmount = amount;
+      } else {
+        const quotation = await apisdk.protocols.paraswapv5.getSwapTokenQuotation(chainId, {
+          input: { token: sourceToken, amount },
+          tokenOut: baseToken.wrapped,
+          slippage,
+        });
+
+        if (!slippage || slippage === 0) {
+          targetTokenAmount = quotation.output.amount;
+        } else {
+          targetTokenAmount = new BigNumberJS(quotation.output.amount)
+            .multipliedBy(common.BPS_BASE - slippage)
+            .div(common.BPS_BASE)
+            .toString();
+        }
```

Comment:
```suggestion
        targetTokenAmount = slippage
          ? quotation.output.setWei(common.calcSlippage(quotation.output.amountWei, slippage)).amount
          : quotation.output.amount;
```
When performing division calculations involving token amounts, it's important to consider the token decimals. The `common.TokenAmount` class will take care of this aspect.